packages/design-picker/src/components/mshots-image/index.tsx

```diff
@@ -35,57 +45,76 @@ export function mshotsUrl( url: string, options: MShotsOptions, count = 0 ): str
 
 const MAXTRIES = 10;
 
+// https://stackoverflow.com/a/60458593
+const useMshotsUrl = ( src: string, options: MShotsOptions ) => {
+	const [ loadedSrc, setLoadedSrc ] = useState( '' );
+	const [ count, setCount ] = useState( 0 );
+
+	useEffect( () => {
+		const img = new Image();
+		const srcUrl = mshotsUrl( src, options, count );
+		img.src = srcUrl;
+		img.onload = () => {
+			// Detect default image (Don't request a 400x300 image).
+			if ( img.naturalWidth !== 400 || img.naturalHeight !== 300 ) {
+				setLoadedSrc( srcUrl );
+			} else {
+				// Only refresh 10 times
+				count < MAXTRIES &&
+					// Triggers a target.src change with increasing timeouts
+					setTimeout( () => setCount( count + 1 ), count * 500 );
+			}
+		};
```

Comment:
Calling `setLoadedSrc` or `setCount` after the `<MShotsImage>` component has unmounted is a no-no, and the `useMshotsUrl` hook isn't dealing with that situation.

You can see this if you open the browser console, load the designs page with the cache buster flag, and before the thumbnails finish loading click "Go back". You'll get a React error message in the console about trying to change state after unmount.

This can be fixed with some careful cleanup.
```suggestion
		const img = new Image();
		const srcUrl = mshotsUrl( src, options, count );
		let timeoutId: number;
		img.src = srcUrl;
		img.onload = () => {
			// Detect default image (Don't request a 400x300 image).
			if ( img.naturalWidth !== 400 || img.naturalHeight !== 300 ) {
				setLoadedSrc( srcUrl );
			} else if ( count < MAXTRIES ) {
				// Only refresh 10 times
				// Triggers a target.src change with increasing timeouts
				timeoutId = setTimeout( () => setCount( count + 1 ), count * 500 );
			}
		};
		
		return () => {
			img.onload = null;
			clearTimeout( timeoutId );
		};
```

(github isn't nice about side-by-side diffs, my changes are I've added a `timeoutId` variable and am returning a cleanup function from the `useEffect` callback)

This should fix up the React errors.

Reply:
Thanks for the patch! https://github.com/Automattic/wp-calypso/pull/51261/commits/c93e4ef8a604df2834f373c67c0f4f7155f42cbb